### PR TITLE
Net first pass

### PR DESCRIPTION
**Pull request by rpmoore**
Committing now since there was some code that I needed that is common between c# and java.  Will continue to work on adding unit tests to verify that all the arguments are getting generated correctly.  General framework for how special casing will work is in place.

### CODE REVIEW

File: ds3-autogen-net/src/main/java/com/spectralogic/ds3autogen/net/NetCodeGenerator.java [hunk 1]
```diff
@@ -52,7 +57,7 @@ public NetCodeGenerator() {
 
     @Override
     public void generate(final Ds3ApiSpec spec, final FileUtils fileUtils, final Path destDir) throws IOException {
```
Comment (RachelTucker) [inline ds3-autogen-net/src/main/java/com/spectralogic/ds3autogen/net/NetCodeGenerator.java]: Extra space

File: ds3-autogen-net/src/main/java/com/spectralogic/ds3autogen/net/NetHelper.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    public static String constructor(final ImmutableList<Arguments> args) {
+        return sortConstructorArgs(args)
+                .stream()
+                .map(i -> getType(i) + " " + uncapFirst(i.getName()))
+                .collect(Collectors.joining(", "));
+    }
+
+    /**
+     * Gets the type of an argument, converting the type from Contract type to Java type.
```
Comment (RachelTucker) [inline ds3-autogen-net/src/main/java/com/spectralogic/ds3autogen/net/NetHelper.java]: Why is the NetHelper converting from Contract type to Java type? Is this a typo?

File: ds3-autogen-net/src/main/java/com/spectralogic/ds3autogen/net/NetHelper.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                return "null";
+            case "string":
+                return capFirst(arg.getName());
+            case "double":
+            case "integer":
+            case "int":
+            case "long":
+                return  capFirst(arg.getName()) + ".ToString()";
+            default:
+                return uncapFirst(arg.getName()) + ".toString()";
```
Comment (RachelTucker) [inline ds3-autogen-net/src/main/java/com/spectralogic/ds3autogen/net/NetHelper.java]: Should this be a capital T: ".ToString()"?

File: ds3-autogen-net/src/main/resources/tmpls/net/common/constructor.ftl [hunk 4]
```diff
@@ -0,0 +1,5 @@
+        public ${name}(${netHelper.constructor(requiredArgs)}) {
```
Comment (RachelTucker) [inline ds3-autogen-net/src/main/resources/tmpls/net/common/constructor.ftl]: Will this include required args that have "void" type?

File: ds3-autogen-java/src/main/java/com/spectralogic/ds3autogen/java/helpers/JavaHelper.java [hunk 5]
```diff
@@ -284,7 +284,7 @@ public static String convertType(final Element element) throws IllegalArgumentEx
     }
 
     /**
-     * Creates a comma separated list of Elements.  This is used in the Java model generation.
+     * Creates a comma separated list of Elements.  This is used in the Java requestmodels generation.
```
Comment (RachelTucker) [inline ds3-autogen-java/src/main/java/com/spectralogic/ds3autogen/java/helpers/JavaHelper.java]: This is not used by requests, but used when generating models from Types

File: ds3-autogen-java/src/main/java/com/spectralogic/ds3autogen/java/helpers/JavaHelper.java [hunk 6]
```diff
@@ -297,7 +297,7 @@ public static String getModelConstructorArgs(final ImmutableList<Element> elemen
     }
 
     /**
-     * Sorts Elements. Used in Java model generation.
+     * Sorts Elements. Used in Java requestmodels generation.
```
Comment (RachelTucker) [inline ds3-autogen-java/src/main/java/com/spectralogic/ds3autogen/java/helpers/JavaHelper.java]: same as above

File: ds3-autogen-java/src/main/java/com/spectralogic/ds3autogen/java/helpers/JavaHelper.java [hunk 7]
```diff
@@ -313,7 +313,7 @@ public static String getModelConstructorArgs(final ImmutableList<Element> elemen
     }
 
     /**
-     * Creates a comma separated list of enum constants. Used in Java model generation.
+     * Creates a comma separated list of enum constants. Used in Java requestmodels generation.
```
Comment (RachelTucker) [inline ds3-autogen-java/src/main/java/com/spectralogic/ds3autogen/java/helpers/JavaHelper.java]: This is used when generating enum models, not requests

File: ds3-autogen-net/src/main/resources/tmpls/net/common/constructor.ftl (hunk 4 above)
Comment (rpmoore) [inline ds3-autogen-net/src/main/resources/tmpls/net/common/constructor.ftl]: Not here in the template.  I'll be removing void types in the model generator.

File: ds3-autogen-net/src/main/java/com/spectralogic/ds3autogen/net/model/response/BaseResponse.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
+package com.spectralogic.ds3autogen.net.model.response;
```
Comment (RachelTucker) [inline ds3-autogen-net/src/main/java/com/spectralogic/ds3autogen/net/model/response/BaseResponse.java]: missing copyright

File: ds3-autogen-net/src/main/java/com/spectralogic/ds3autogen/net/util/GeneratorUtils.java [hunk 9]
```diff
@@ -0,0 +1,117 @@
+package com.spectralogic.ds3autogen.net.util;
```
Comment (RachelTucker) [inline ds3-autogen-net/src/main/java/com/spectralogic/ds3autogen/net/util/GeneratorUtils.java]: missing copyright

File: ds3-autogen-utils/src/main/java/com/spectralogic/ds3autogen/utils/Ds3RequestUtils.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
+package com.spectralogic.ds3autogen.utils;
```
Comment (RachelTucker) [inline ds3-autogen-utils/src/main/java/com/spectralogic/ds3autogen/utils/Ds3RequestUtils.java]: missing copyright

File: ds3-autogen-utils/src/main/java/com/spectralogic/ds3autogen/utils/ProjectConstants.java [hunk 11]
```diff
@@ -0,0 +1,10 @@
+package com.spectralogic.ds3autogen.utils;
```
Comment (RachelTucker) [inline ds3-autogen-utils/src/main/java/com/spectralogic/ds3autogen/utils/ProjectConstants.java]: missing copyright

**Pull request merged**